openedx/core/lib/block_cache/transformer.py

```diff
@@ -0,0 +1,123 @@
+"""
+This module provides the abstract base class for all Block Structure
+Transformers.
+"""
+from abc import abstractmethod
+
+
+class BlockStructureTransformer(object):
+    """
+    Abstract base class for all block structure transformers.
+    """
+
+    # All Transformers are expected to maintain a VERSION class
+    # attribute.  While the value for the base class is set to 0,
+    # the value for each concrete transformer should be 1 or higher.
+    #
+    # A transformer's version attribute is used by the block_cache
+    # framework in order to determine whether any collected data for a
+    # transformer is outdated.  When a transformer's data is collected
+    # and cached, it's version number at the time of collection is
+    # stored along with the data.  That version number is then checked
+    # at the time of accessing the collected data (during the transform
+    # phase).
+    #
+    # The version number of a Transformer should be incremented each
+    # time the implementation of its collect method is updated such that
+    # its collected data is changed.
+    #
+    VERSION = 0
+
+    @classmethod
+    def name(cls):
+        """
+        Unique identifier for the transformer's class.  It is used to
+        identify the transformer's cached data. So do not override or
+        modify without consideration of backward compatibility with
+        previously collected data.
+        """
+        return cls.__name__
```

Comment:
This returns just the name of the class.  Not the location of the class.  So it still allows the class to be moved to other modules/files at any later point.  It can always be overriden by subclasses if they choose.